tests/integration/tests/test_gateway.py

```diff
@@ -37,8 +38,10 @@ def test_gateway(session_instance: harness.Instance):
         ]
     )
 
+    # Get gateway node port
+    gateway_http_port = None
     util.stubbornly(retries=5, delay_s=2).on(session_instance).until(
-        lambda p: "cilium-gateway-my-gateway" in p.stdout.decode()
+        lambda p: "my-gateway" in p.stdout.decode()
     ).exec(["k8s", "kubectl", "get", "service", "-o", "json"])
```

Comment:
this code here can be dangerous. we repeatedly get the list of services until `my-gateway` appears on the text. this is right, as the service might take some time to appear (perhaps the 5 retries with 2 seconds backoff is also very short).

However, we then retrieve them again to parse the service info and retrieve the nodePort. Note that this nodePort might not be available upon service creation, so there is a race here: If the test retrieves the service before the ports are configured, then our test will flake.

Instead, we should do all of the logic inside the `util.stubbornly`. Currently, it's using a lambda, but we should probably write this as a proper function instead, then do the retries until we get the nodePort we were after.

If we don't get the nodePort in time, that's OK, the timeout here will tell us about the failure